.github/workflows/connector-ci-tests.yml

```diff
@@ -43,14 +54,25 @@ jobs:
           echo "jvm_connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json --java)" | tee -a $GITHUB_OUTPUT
           echo "non_jvm_connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json --no-java)" | tee -a $GITHUB_OUTPUT
 
+      - name: Set JVM Runner Type
+        id: set-jvm-runner-type
+        run: |
+          # Default to ubuntu-24.04 for all connectors
+          echo "jvm-runner-type=ubuntu-24.04" >> $GITHUB_OUTPUT
+          if [[ "${{ github.repository }}" == "airbytehq/airbyte" ]]; then
+            # Within our own repo, we can use the custom (larger) runner
+            echo "jvm-runner-type=linux-24.04-large" | tee -a "$GITHUB_OUTPUT"
+          fi
```

Comment:
I wrongly accepted earlier that dynamic `runs-on` clauses runners were not supported. In fact, this works fine.